### PR TITLE
allow non-admin users to interact with user/groups

### PR DESCRIPTION
**Pull request by switj**
If a user is not a super admin they will fail to read Admin roles. This
change will continue with the assumption that there aren't any admin
roles.

### CODE REVIEW

**Comment by switj** [comment]
Hello okta provider maintainers!

This PR addresses an issue that we're facing while trying to use terraform to manage okta resources using a non super admin api key.

Is this something that would be beneficial to merge back to upstream?

I'd appreciate any feedback/suggestions/comments.

Thanks

File: okta/user.go [hunk 1]
```diff
@@ -346,18 +346,18 @@ func populateUserProfile(d *schema.ResourceData) *okta.UserProfile {
...
-	if err != nil {
-		return err
-	}
-
-	roleTypes := make([]interface{}, 0)
-	for _, role := range roles {
-		roleTypes = append(roleTypes, role.Type)
+	// only super admin can list roles if this fails assume the user has no
+	// admin roles
+	if err == nil {
```
Comment (quantumew) [inline okta/user.go]: Love this idea. I think I would tweak this a bit. We should more specifically check the error to make sure it is indeed a permissions error. If it isn't, return the error. If it is, log a warning and not sync `admin_roles` in state.

I can merge this down to a branch that isn't forked, feel free to make that change if you get to it first. Thanks for the PR!

Comment (switj) [reply okta/user.go]: thanks for the feedback. I've taken a second attempt at it.

Comment (quantumew) [reply okta/user.go]: Great work!

**Pull request merged**